src/test/regress/expected/view_propagation.out

```diff
@@ -0,0 +1,320 @@
+HINT:  Distribute "table view_table_1" first to distribute "view prop_view_1"
+SELECT create_distributed_table('view_table_1', 'id');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+CREATE OR REPLACE VIEW prop_view_1 AS
+    SELECT * FROM view_table_1;
+-- Try to create view depending local table, then try to recreate it after making the table reference table
+CREATE TABLE view_table_2(id int PRIMARY KEY, val_1 text);
+CREATE VIEW prop_view_2 AS
+    SELECT view_table_1.id, view_table_2.val_1 FROM view_table_1 INNER JOIN view_table_2
+    ON view_table_1.id = view_table_2.id;
+WARNING:  "view prop_view_2" has dependency to "table view_table_2" that is not in Citus' metadata
+DETAIL:  "view prop_view_2" will be created only locally
+HINT:  Distribute "table view_table_2" first to distribute "view prop_view_2"
+SELECT create_reference_table('view_table_2');
+ create_reference_table
+---------------------------------------------------------------------
+
+(1 row)
+
+CREATE OR REPLACE VIEW prop_view_2 AS
+    SELECT view_table_1.id, view_table_2.val_1 FROM view_table_1 INNER JOIN view_table_2
+    ON view_table_1.id = view_table_2.id;
+-- Try to create view depending local table, then try to recreate it after making the table citus local table
+CREATE TABLE view_table_3(id int, val_1 text);
+CREATE VIEW prop_view_3 AS
+    SELECT * FROM view_table_1 WHERE id IN
+    (SELECT view_table_2.id FROM view_table_2 INNER JOIN view_table_3 ON view_table_2.id = view_table_3.id);
+WARNING:  "view prop_view_3" has dependency to "table view_table_3" that is not in Citus' metadata
+DETAIL:  "view prop_view_3" will be created only locally
+HINT:  Distribute "table view_table_3" first to distribute "view prop_view_3"
+SELECT 1 FROM citus_add_node('localhost', :master_port, groupid=>0);
+NOTICE:  Replicating reference table "view_table_2" to the node localhost:xxxxx
+NOTICE:  localhost:xxxxx is the coordinator and already contains metadata, skipping syncing the metadata
+ ?column?
+---------------------------------------------------------------------
+        1
+(1 row)
+
+ALTER TABLE view_table_3
+ADD CONSTRAINT f_key_for_local_table
+FOREIGN KEY(id)
+REFERENCES view_table_2(id);
+CREATE OR REPLACE VIEW prop_view_3 AS
+    SELECT * FROM view_table_1 WHERE id IN
+    (SELECT view_table_2.id FROM view_table_2 INNER JOIN view_table_3 ON view_table_2.id = view_table_3.id);
+-- Try to create view depending on PG metadata table
+CREATE VIEW prop_view_4 AS
+    SELECT * FROM pg_stat_activity;
+-- Try to create view depending on Citus metadata table
+CREATE VIEW prop_view_5 AS
+    SELECT * FROM citus_dist_stat_activity;
+-- Try to create table depending on a local table from another schema, then try to create it again after distributing the table
+CREATE SCHEMA view_prop_schema_inner;
+SET search_path TO view_prop_schema_inner;
+-- Create local table for tests below
```

Comment:
hmm, we can convert Citus local tables to dist/ref tables, where views are lost. So, we should probably re-create these. We are even considering making `alter system set citus.use_citus_managed_tables to on;` the default somewhere in the not so far future

```SQL
-- add coordinator
SELECT set_coodinator_host('localhost', 5432);


-- enable Citus managed tables
alter system set citus.use_citus_managed_tables to on;
SELECT pg_reload_conf();

CREATE TABLE local(a int);
CREATE VIEW v1 AS SELECT * FROM local;


-- views are created on the workers
SELECT run_command_on_workers($$SELECT 'v1'::regclass$$);
┌────────────────────────┐
│ run_command_on_workers │
├────────────────────────┤
│ (localhost,9700,t,v1)  │
│ (localhost,9701,t,v1)  │
│ (localhost,9702,t,v1)  │
│ (localhost,9703,t,v1)  │
│ (localhost,9704,t,v1)  │
│ (localhost,9705,t,v1)  │
│ (localhost,9706,t,v1)  │
└────────────────────────┘
(7 rows)

-- convert Citus local to dist
SELECT create_distributed_table('local', 'a');



-- ops, views on the workers are gone
SELECT run_command_on_workers($$SELECT 'v1'::regclass$$);
┌─────────────────────────────────────────────────────────────┐
│                   run_command_on_workers                    │
├─────────────────────────────────────────────────────────────┤
│ (localhost,9700,f,"ERROR:  relation ""v1"" does not exist") │
│ (localhost,9701,f,"ERROR:  relation ""v1"" does not exist") │
│ (localhost,9702,f,"ERROR:  relation ""v1"" does not exist") │
│ (localhost,9703,f,"ERROR:  relation ""v1"" does not exist") │
│ (localhost,9704,f,"ERROR:  relation ""v1"" does not exist") │
│ (localhost,9705,f,"ERROR:  relation ""v1"" does not exist") │
│ (localhost,9706,f,"ERROR:  relation ""v1"" does not exist") │
└─────────────────────────────────────────────────────────────┘
(7 rows)
```

Reply:
As we've discussed online. I'll open subsequent PR to handle views once the table type changes.